src/generics.jl

```diff
@@ -68,27 +86,41 @@ julia> W * W'
  0.0  1.0
 ```
 """
-whiten(a::AbstractPDMat, x::StridedVecOrMat) = whiten!(similar(x), a, x)
-unwhiten(a::AbstractPDMat, x::StridedVecOrMat) = unwhiten!(similar(x), a, x)
+whiten(a::AbstractMatrix, x::AbstractVecOrMat) = whiten!(similar(x), a, x)
```

Comment:
Unrelated, but would probably be good to not fall back on the in-place methods here to avoid e.g. creating `MArray` from `SArray`s (IIRC we already did this for `PDMat` etc.).